### PR TITLE
Fix function config and remove createFunction API

### PR DESCRIPTION
**Pull request by kamilafsar**
Few fixes for on config:

- Remove the `createFunction` API. We don't really need it since we're removing config on functions.
- Remove deployment config from ServiceConfig. For now you can only configure middleware on your service. Next step is to add cors config.
- Fixes leftovers from an earlier change (moving context to be the first parameter in the manifest file)

### CODE REVIEW

File: packages/cli/src/components/ProjectStatus/ClientProjectStatus.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ export default function ClientProjectStatus({
         break
 
       case "BUILD_FAILED":
-        setEvent(["error", "Could not build client"])
+        setEvent(["error", "Could not build client" + event.errorMessage])
```
Comment (Jpunt) [inline packages/cli/src/components/ProjectStatus/ClientProjectStatus.tsx:61]:
```suggestion
        setEvent(["error", "Could not build client: " + event.errorMessage])
```

File: packages/core/src/generateAppDeclarationFile.test.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ describe("generateAppDeclarationFile", () => {
     test("should parse middleware", () => {
       const parsedApp = generate(
         createTestProgram(`
-        type NextFunction<T = void> = T extends void
+        type SamenNextFunction<T = void> = T extends void
```
Comment (Jpunt) [inline packages/core/src/generateAppDeclarationFile.test.ts:24]: Ah, nice 👍

**Review by Jpunt — APPROVED** [review]
As always, great work mate 👏

File: packages/cli/src/components/ProjectStatus/ClientProjectStatus.tsx (hunk 1 above)
Comment (kamilafsar) [reply packages/cli/src/components/ProjectStatus/ClientProjectStatus.tsx]: Oops, it was not my intention to commit this... reverted it.

File: packages/core/src/generateAppDeclarationFile.test.ts (hunk 2 above)
Comment (kamilafsar) [reply packages/core/src/generateAppDeclarationFile.test.ts:24]: Yes.... but now I realised we need to rework docs a bit :D

**Comment by kamilafsar** [comment]
> As always, great work mate 👏

Thanks mate 😊

**Pull request merged**